### PR TITLE
Add a dedicated library for json.

### PR DESCRIPTION
**Pull request by siarheishut**

### CODE REVIEW

File: lib/json/src/json.cpp
Comment (foxfrikses) [inline lib/json/src/json.cpp:1]: Send this as a separate commit, and add tests for output.

File: lib/json/include/json.h [hunk 1]
```diff
@@ -1,5 +1,5 @@
-#ifndef ROOT_MANAGER_SRC_JSON_JSON_H_
-#define ROOT_MANAGER_SRC_JSON_JSON_H_
+#ifndef JSON_JSON_H_
+#define JSON_JSON_H_
 
 #include <istream>
 #include <ostream>
```
Comment (foxfrikses) [inline lib/json/include/json.h:5]: remove rm namespace

File: CMakeLists.txt [hunk 2]
```diff
@@ -12,29 +12,30 @@ FetchContent_Declare(
...
         tests/bus_manager_test.cpp
-        tests/json_test.cpp
         tests/test_utils.cpp)
 
-target_link_libraries(route_manager_tests GTest::gtest_main)
+target_link_libraries(route_manager_tests GTest::gtest_main json)
 target_include_directories(route_manager_tests PUBLIC . tests)
 gtest_discover_tests(route_manager_tests)
 # tests end
+
```
Comment (foxfrikses) [inline CMakeLists.txt:41]: extra line

File: lib/json/tests/tests.cpp
Comment (foxfrikses) [inline lib/json/tests/tests.cpp:1]: Delete these tests from rm.

**Review by foxfrikses — CHANGES_REQUESTED** [review]

File: CMakeLists.txt (hunk 2 above)
Comment (siarheishut) [reply CMakeLists.txt:41]: Done.

File: lib/json/include/json.h (hunk 1 above)
Comment (siarheishut) [reply lib/json/include/json.h:5]: Done.

File: lib/json/src/json.cpp
Comment (siarheishut) [reply lib/json/src/json.cpp:1]: Will be done.

File: lib/json/tests/tests.cpp
Comment (siarheishut) [reply lib/json/tests/tests.cpp:1]: Done.

**Review by foxfrikses — APPROVED** [review]
LGTM

**Pull request merged**